### PR TITLE
Allow throughput with defaulted GP3 volume type

### LINKED ISSUE #1583: Throughput parameter doesn't work
/kind bug

**What happened?**
Throughput parameter on StorageClass not being implemented for dynamic provisioning.

**What you expected to happen?**
I was using a `gp3` EBS type and specified throughput as 250Mb/s, and it came up as 125Mb/s.

**How to reproduce it (as minimally and precisely as possible)?**
Define a StorageClass as follows:
```
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: somename
parameters:
  encrypted: "true"
  iops: "5000"
  kmsKeyId: "kmskeyId"
  throughput: "250"
provisioner: ebs.csi.aws.com
reclaimPolicy: Delete
volumeBindingMode: WaitForFirstConsumer
```
  
**Anything else we need to know?**:

**Environment**
- Kubernetes version (use `kubectl version`): 
```
Client Version: v1.24.10-eks-48e63af
Kustomize Version: v4.5.4
Server Version: v1.25.8-eks-ec5523e
```
- Driver version: v1.18.0

### PR DESCRIPTION
**Pull request by ConnorJC3**
Fixes #1583 and adds a test case for the scenario

### CODE REVIEW

File: pkg/cloud/cloud.go [hunk 1]
```diff
@@ -393,7 +393,7 @@ func (c *cloud) CreateDisk(ctx context.Context, volumeName string, diskOptions *
 	if iops > 0 {
 		requestInput.Iops = aws.Int64(iops)
 	}
-	if throughput > 0 && diskOptions.VolumeType == VolumeTypeGP3 {
+	if throughput > 0 && createType == VolumeTypeGP3 {
```
Comment (torredil) [inline pkg/cloud/cloud.go:396]: Is this check `createType == VolumeTypeGP3` necessary? throughput will only ever be > 0 if `createType = VolumeTypeGP3` or `createType == ""`: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/dd719c12ef9cbf8e2b86152003b3afb8fa0057e9/pkg/cloud/cloud.go#L332

**Review by torredil — COMMENTED** [review]
Is this check `createType == VolumeTypeGP3` necessary?

File: pkg/cloud/cloud.go (hunk 1 above)
Comment (ConnorJC3) [reply pkg/cloud/cloud.go:396]: That's not true, we don't check the volume type when pass through the throughput in controller.go

**Review by torredil — APPROVED** [review]
/approve

File: pkg/cloud/cloud_test.go [hunk 2]
```diff
@@ -1961,8 +1980,8 @@ func (m *eqCreateVolumeMatcher) Matches(x interface{}) bool {
 	if input == nil {
 		return false
 	}
-	// Compare only IOPS for now
-	ret := reflect.DeepEqual(m.expected.Iops, input.Iops)
+	// TODO: Check all inputs
+	ret := reflect.DeepEqual(m.expected.Iops, input.Iops) && reflect.DeepEqual(m.expected.Throughput, input.Throughput)
```
Comment (rdpsin) [inline pkg/cloud/cloud_test.go:1984]: Can we use `cmp.Diff` here?

**Review by hanyuel — APPROVED** [review]

**Pull request merged**